test/TestMatchError.py

```diff
@@ -0,0 +1,44 @@
+from ansiblelint.errors import MatchError
+from ansiblelint.rules import AnsibleLintRule
+from ansiblelint.rules.AlwaysRunRule import AlwaysRunRule
+from ansiblelint.rules.BecomeUserWithoutBecomeRule import BecomeUserWithoutBecomeRule
+
+import pytest
+
+
+def test_matcherror_compare():
+    assert MatchError("foo") == MatchError("foo")
+
+
+class AnsibleLintRuleWithStringId(AnsibleLintRule):
+    id = "200"
+
+
+def test_matcherror_invalid():
+    """Ensure that MatchError requires message or rule."""
+    expected_err = r"^MatchError\(\) missing a required argument: one of 'message' or 'rule'$"
+    with pytest.raises(TypeError, match=expected_err):
+        MatchError()
+
+
+@pytest.mark.parametrize(
+    ('a', 'b'), (
+        # sorting by message
+        (MatchError("z"), MatchError("a")),
+        # filenames takes priority in sorting
+        (MatchError("a", filename="b"), MatchError("a", filename="a")),
+        # rule id 501 > rule id 101
+        (MatchError(rule=BecomeUserWithoutBecomeRule), MatchError(rule=AlwaysRunRule)),
+        # rule id "200" > rule id 101
+        (MatchError(rule=AnsibleLintRuleWithStringId), MatchError(rule=AlwaysRunRule))
+    ))
+class TestMatchErrorCompare():
+
+    def test_lt(self, a, b):
+        assert b < a
+
+    def test_gt(self, a, b):
+        assert a > b
+
+    def test_eq(self, a, b):
+        assert a != b
```

Comment:
```suggestion
parametrize_match_error = pytest.mark.parametrize(
    ('left_match_error', 'right_match_error'),
    (
        # sorting by message
        (MatchError("z"), MatchError("a")),

        # filename takes priority in sorting
        (MatchError("a", filename="b"), MatchError("a", filename="a")),

        # rule id 501 > rule id 101
        (MatchError(rule=BecomeUserWithoutBecomeRule), MatchError(rule=AlwaysRunRule)),

        # rule id "200" > rule id 101
        (MatchError(rule=AnsibleLintRuleWithStringId), MatchError(rule=AlwaysRunRule)),

        # line should be taken into account
        (MatchError("a", line="b"), MatchError("a", line="a")),
    ),
)


@parametrize_match_error
def test_match_error_less_than(left_match_error, right_match_error):
    """Check 'less than' protocol implementation in MatchError."""
    assert left_match_error < right_match_error


@parametrize_match_error
def test_match_error_greater_than(left_match_error, right_match_error):
    """Check 'greater than' protocol implementation in MatchError."""
    assert left_match_error > right_match_error


@parametrize_match_error
def test_match_error_not_equal(left_match_error, right_match_error):
    """Check 'not equals' protocol implementation in MatchError."""
    assert left_match_error != right_match_error
```

Reply:
@ssbarnea ^

Reply:
I see no compelling reason not to use classes, especially as this is not the first place to do so.

Can we please stop the nitpicking and stick to things that really matter?

For the dimension of this change it received an overwhelming amount of change requests, many of them even contradictory.

The review process is not about forcing the OP to write the change and exactly how the reviewer would have written it. All this nitpicking is draining and discouraging me from contributing.

Reply:
I don't think that having classes is a good idea. Also, "not the first place" is a bad excuse. Those things are legacy unittest-based modules mostly.

FWIW, if you want to keep the class, feel free but you must add docstrings and proper variable names. I won't approve a/b.
Also, add a param with lines (see the code suggestion above). I requested this to be present in tests several times already.

Reply:
^ the request still stands

Reply:
Implemented var rename plus docstrings. even if I doubt the var names really add value for final reports or readability of the two-line test functions.